### DISCUSSION TITLE
Can preset-commonmark and preset-gfm be used together?

### DISCUSSION THREAD

**Discussion by 9vek** [Milkdown/milkdown#410] [Q&A] [answered]
I used to think the `preset-commonmark` and `preset-gfm` are complement to each other, I use both of them before. But after trying upgrading them from `5.3.0` to `5.5.0`, I got this error: 

```
RangeError: Adding different instances of a keyed plugin (MILKDOWN_PLUGIN_ID$)
    at state.js:50:15
    at Array.forEach (<anonymous>)
    at new Configuration2 (state.js:48:26)
    at Function.create3 [as create] (state.js:184:19)
    at editor-state.ts:65:35
    at async Promise.all (index 54)
    at async _Editor.create (editor.ts:86:35)
```

So now I have to choose to keep `preset-commonmark` **or** `preset-gfm`, I wonder is it not a bug? Is that mean the two plugins are confilct to each other?

**Comment by Saul-Mirone** [comment, marked as answer]
preset-gfm has already includes everything in preset-commonmark.

**Reply by 9vek** [reply]
谢谢嗷

**Reply by 9vek** [reply]
By the way, I used to write this kind of code:

```ts
import { heading } from '@milkdown/preset-commonmark'
export const myHeading = heading.extend((original, utils, options) => {
  // ...
}
```

And I realized that I cannot simply migrate it like that:
```ts
import { heading } from '@milkdown/preset-gfm'
export const myHeading = heading.extend((original, utils, options) => {
  // ...
}
```
That would give an error, is that mean preset-gfm does not allow the extend grammar like this?

**Reply by Saul-Mirone** [reply]
I think extend should work as usual, what's the error you got?

**Reply by 9vek** [reply]
> I think extend should work as usual, what's the error you got?

I cannot recreate this error, it now works fine😄  the error I met that day was possibly caused by a temporary unawareness of my IDE.
thank you for your reply.
